### ISSUE TITLE
When *ring* feature is enabled, use it for RSA and other supported DNSSec proofs 

### ISSUE THREAD

**Issue by bluejekyll** [hickory-dns/hickory-dns#180]
As of now, all RSA and ECDSA proofs are performed with OpenSSL. It would be good to remove the dependency on OpenSSL entirely but allow RSA to be proved with the *ring* library.

**Comment by briansmith** [comment]
I started working on this in #194, #195, #196, and #198.

I found that `create_sig0_ready_client`, `test_rrset_tbs`, `create_secure_example`, `test_sign_and_verify_message_sig0()`, and `test_sign_and_verify_rrset()` are still generating their own private keys, which makes it hard to adapt them to work with *ring*, since *ring* doesn't do keygen itself. keygen is really a server-side feature, not a client-side feature, so it would be better to eliminate the need for keygen in these client-side tests. However, I probably won't have time to do *that* conversion myself.

However, once that conversion is done, I will be able to finish the work to make the entire client side (verification side) work with *ring*, including RSA, ECDSA, Ed25519.

Note, that *ring* will reject small public RSA keys (less than 2048 bits), but I think that this is the best behavior anyway. People who want legacy stuff to work can use legacy crypto libraries.

**Comment by briansmith** [comment]
Also #199.

**Comment by briansmith** [comment]
Removing key generation from the tests is already filed as a separate issue: #134.

**Comment by briansmith** [comment]
Now also RSA verification: PR #203.

**Comment by briansmith** [comment]
Also TrustAnchor: #204.

**Comment by briansmith** [comment]
Ideally a *ring*-enabled client would build and be testable without OpenSSL. One way of accelerating that process is by implementing the change in #205, which proposes moving signing/private-key operations to the server crate and using pregenerated responses for testing signature verification in the client crate. IMO it makes more sense to do this now than to just fix #134, because keygen isn't the only server-specific functionality that has OpenSSL dependencies, and because #205 seems like the right thing to do, long-term.

**Comment by briansmith** [comment]
This was done and this should be closed. #134 can track making the tests work when *ring* is used and OpenSSL isn't available.

**Comment by briansmith** [comment]
*ring* now has ECDSA signing support, so ECDSA signing should be made to use *ring* when *ring* is enabled: https://github.com/briansmith/ring/issues/207#issuecomment-393662940

**Comment by bluejekyll** [comment]
I just saw that land! Awesome!
